common/common.cpp

```diff
@@ -836,7 +836,12 @@ bool gpt_params_find_arg(int argc, char ** argv, const std::string & arg, gpt_pa
             invalid_param = true;
             return true;
         }
-        params.n_gpu_layers = std::stoi(argv[i]);
+        std::string argValue = argv[i];
+        if (argValue == "auto" || argValue == "a") {
```

Comment:
I agree it can be a breaking change, but I would prefer to have this approach as the default. i.e. if `-ngl` is not passed: automatically offload the maximum possible layers to VRAM.

Reply:
Could be. If someone doesn't want that, they could simply `-ngl 0` or just not compile with GPU args passed.

Reply:
Yes but this is just my personal point of view. @ggerganov or @slaren would have a better global view